shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java

```diff
@@ -670,6 +672,22 @@ public Map<String, String> getLogicAndActualTablesFromBindingTable(final String
                 .orElseGet(Collections::emptyMap);
     }
     
+    /**
+     * Get logic table and actual table.
+     *
+     * @param routeUnit Route unit
+     * @param availableLogicBindingTables available logic binding table names
+     * @return key is logicTable name, values is actualTable belong to this data source
+     */
+    public Map<String, String> getLogicAndActualTablesFromRouteUnit(final RouteUnit routeUnit, final Collection<String> availableLogicBindingTables) {
+        Map<String, String> result = new HashMap<>(availableLogicBindingTables.size(), 1);
+        for (RouteMapper each : routeUnit.getTableMappers()) {
+            result.put(each.getLogicName().toLowerCase(), each.getActualName());
+            result.putAll(this.getLogicAndActualTablesFromBindingTable(routeUnit.getDataSourceMapper().getLogicName(), each.getLogicName(), each.getActualName(), availableLogicBindingTables));
```

Comment:
I don't know why is do that.  Get last actualTableName from bindingTable ?

Reply:
Where is the code come from?

Reply:
What is the relationship about this method and ShardingRule? Why put the method here?

Reply:
https://github.com/apache/shardingsphere/blob/d0ca7711d4050f2d028d97538dfbec4f7cb50ccb/shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rewrite/token/pojo/TableToken.java

Reply:
I just remove `getLogicAndActualTables` to ShardingRule

Reply:
> What is the relationship about this method and ShardingRule? Why put the method here?

yes, it's probably not a good idea.